store/rootmulti/store.go

```diff
@@ -389,15 +416,29 @@ func (rs *Store) pruneStores() {
 			// it to get the underlying IAVL store.
 			store = rs.GetCommitKVStore(key)
 
-			if err := store.(*iavl.Store).DeleteVersions(rs.pruneHeights...); err != nil {
+			if err := store.(*iavl.Store).DeleteVersions(pruneHeights...); err != nil {
 				if errCause := errors.Cause(err); errCause != nil && errCause != iavltree.ErrVersionDoesNotExist {
 					panic(err)
 				}
 			}
 		}
 	}
 
-	rs.pruneHeights = make([]int64, 0)
+	// remove pruned heights from rs.pruneHeights
+	// new rs.pruneHeights will be updated by next flushMedataData cycle
+	rs.pruneLock.Lock()
```

Comment:
We should add this call of `rs.pruneLock.Lock()` before appending new `pruneHeight`.